### PR TITLE
Added progress bars for MH and NUTS

### PR DESCRIPTION
**Pull request by jacobaustin123**

### CODE REVIEW

File: include/autoppl/algorithm/mh.hpp [hunk 1]
```diff
@@ -64,7 +65,13 @@ inline void mh__(ModelType& model,
     std::uniform_real_distribution unif_sampler(0., 1.);
 
     for (size_t iter = 0; iter < n_sample + warmup; ++iter) {
-        
+        if (iter % ((n_sample + warmup) / 100) == 0) {
```
Comment (JamesYang007) [inline include/autoppl/algorithm/mh.hpp]: Seems like we can abstract this out into a logging function - can we actually start another header file in autoppl/logger dir with such function and reuse this function in NUTS as well

**Review by JamesYang007 — CHANGES_REQUESTED** [review]
👯

File: include/autoppl/algorithm/mh.hpp (hunk 1 above)
Comment (jacobaustin123) [reply include/autoppl/algorithm/mh.hpp]: Ok makes sense. How about this?

**Review by JamesYang007 — APPROVED** [review]
🕺

**Pull request merged**